src/compiler/emitter.ts

```diff
@@ -138,14 +138,27 @@ namespace ts {
     /* @internal */
     export function getOutputDeclarationFileName(inputFileName: string, configFile: ParsedCommandLine, ignoreCase: boolean) {
         Debug.assert(!fileExtensionIs(inputFileName, Extension.Dts) && hasTSFileExtension(inputFileName));
-        const relativePath = getRelativePathFromDirectory(rootDirOfOptions(configFile), inputFileName, ignoreCase);
-        const outputPath = resolvePath(configFile.options.declarationDir || configFile.options.outDir || getDirectoryPath(Debug.assertDefined(configFile.options.configFilePath)), relativePath);

```

Comment:
why not use configFilePath anymore?

Reply:
We shouldn't need to map file paths when there is no outDir or declarationDir since the path is going to be same as inputPath (no need to calculate relative paths etc which can result in wrong path when root dir is specified without specifying outdir)